tests/test_gce.py

```diff
@@ -1,143 +1,165 @@
+    assert gce.list_regions() == ['Wonderland']
+
+
+def test_list_zones(gce):
+    gce.compute_client.regions = MockResource({'zones': ['somethingthatIdonotknow/RabbitHole']})
+    gce.compute_client.list_zones = {'zones': ['somethingthatIdonotknow/RabbitHole']}
+    assert gce.list_zones('Oxfordshire') == ['RabbitHole']
 
 
-def test_cleanup_all(monkeypatch):
+def _test_cleanup(gce, resource_type, cleanup_call, dry_run: bool):
     older_than_max_age = (datetime.now(timezone.utc) - timedelta(hours=max_age_hours+1)).strftime("%m/%d/%Y, %H:%M:%S")
     now_age = datetime.now(timezone.utc).strftime("%m/%d/%Y, %H:%M:%S")
 
-    fmi = FakeMockImages([
-        FakeRequest({   # on images().list()
+    resources = MockResource([
+        MockRequest({   # on images().list()
             'items': [
                 {'name': 'keep', 'creationTimestamp': now_age},
                 {'name': 'delete1', 'creationTimestamp': older_than_max_age}
             ], 'id': "id"}),
-        FakeRequest(),    # on images().delete()
-        FakeRequest({   # on images().list_next()
+        MockRequest(),  # on images().delete()
+        MockRequest({   # on images().list_next()
             'items': [
                 {'name': 'keep', 'creationTimestamp': now_age},
                 {'name': 'delete2', 'creationTimestamp': older_than_max_age}
             ], 'id': "id"}),
-        FakeRequest({'error': {'errors': [{'message': 'err message'}]},
+        MockRequest({'error': {'errors': [{'message': 'err message'}]},
                     'warnings': [{'message': 'warning message'}]}),
         None   # on images().list_next()
     ])
 
-    fmd = FakeMockImages([
-        FakeRequest({   # on disks().list()
-            'items': [
-                {'name': 'keep', 'creationTimestamp': now_age},
-                {'name': 'delete_disk1', 'creationTimestamp': older_than_max_age}
-            ], 'id': "id"}),
-        FakeRequest(),    # on disks().delete()
-        FakeRequest({   # on disks().list_next()
-            'items': [
-                {'name': 'keep', 'creationTimestamp': now_age},
-                {'name': 'delete_disk2', 'creationTimestamp': older_than_max_age}
-            ], 'id': "id"}),
-        FakeRequest({'error': {'errors': [{'message': 'err message'}]},
-                    'warnings': [{'message': 'warning message'}]}),
-        None   # on disks().list_next()
-    ])
+    with (
+        patch.object(gce, 'list_regions', return_value=['region1']),
+        patch.object(gce, 'list_zones', return_value=['zone1']),
+    ):
+        gce.compute_client.instances = lambda *args, **kwargs: []
+        setattr(gce.compute_client, resource_type, resources)
+
+        if dry_run:
+            gce.dry_run = True
+            cleanup_call()
+            assert resources.deleted_resources == []
+        else:
+            cleanup_call()
+            assert resources.deleted_resources == ['delete1', 'delete2']
```

Comment:
```suggestion
      gce.dry_run = True
      cleanup_call()
      assert resources.deleted_resources == []
       cleanup_call()
        assert resources.deleted_resources == ['delete1', 'delete2']
```

anyway for every entity you covering both cases , so just unconditionally cover both case and merge all `test_cleanup_<entity>` and `test_cleanup_<entity>_dry_run` into single test

Reply:
> anyway for every entity you covering both cases , so just unconditionally cover both case and merge all `test_cleanup_<entity>` and `test_cleanup_<entity>_dry_run` into single test

I had to use a loop because I can't reuse `resources` .

Reply:
ok